src/pages/index.js

```diff
@@ -3,104 +3,233 @@ import FormValidator from "../components/FormValidator.js";
+const popupWithImage = new PopupWithImage({
+  popupSelector: "#preview-image-modal",
+});
 
-const section = new Section(
+const popupWithDeleteButton = new PopupWithConfirm(
   {
-    items: initialCards,
-    renderer: renderCard,
+    popupSelector: "#delete-card-modal",
   },
-  ".cards__list"
+  openConfirmPopup
 );
 
 const userInfo = new UserInfo({
-  nameElement: ".profile__title",
-  jobElement: ".profile__description",
+  nameElement: ".profile__name",
+  jobElement: ".profile__about",
+  avatarElement: ".profile__avatar",
 });
 
 popupWithEditProfileForm.setEventListeners();
 popupWithAddCardForm.setEventListeners();
 popupWithImage.setEventListeners();
-
-
-
+popupWithDeleteButton.setEventListeners();
+popupWithEditAvatarForm.setEventListeners();
 
 /*Function*/
 
 function renderCard(item, method = "addItem") {
-    const cardElement = getCardElement(item);
-    section[method](cardElement);
+  const cardElement = getCardElement(item);
+  section[method](cardElement);
 }
 
 function getCardElement(cardData) {
-  const card = new Card(cardData, cardSelector, handleImageClick);
+  const card = new Card({
+    cardData,
+    cardSelector,
+    handleDeleteClick: (cardInstance) => openConfirmPopup(cardInstance),
+    handleImageClick,
+    api,
+  });
   return card.getView();
 }
 
 function handleImageClick(data) {
-    popupWithImage.open({ name: data.name, link: data.link });
+  console.log(data.name);
+  popupWithImage.open({ name: data.name, link: data.link });
 }
 
 /*Event Handler*/
 
-function handleProfileEditSubmit(inputValue) {
-    userInfo.setUserInfo({
-        name: inputValue.title,
-        description: inputValue.description,
+function handleProfileEditSubmit(event) {
+  const inputValues = popupWithEditProfileForm.getInputValues(); // Get form values
+
+  if (!inputValues.name || !inputValues.about) {
+    return;
+  }
+
+  popupWithEditProfileForm.setLoadingText(true);
+
+  api
+    .updateUserInfo(inputValues.name, inputValues.about)
+    .then((updatedUser) => {
+      userInfo.setUserInfo(updatedUser);
+      popupWithEditProfileForm.close();
+    })
+    .catch((err) => console.error("Profile update failed:", err))
+    .finally(() => {
+      popupWithEditProfileForm.setLoadingText(false);
+    });
+}
+
+function handleAvatarUpdateSubmit(event) {
+  const inputvalues = popupWithEditAvatarForm.getInputValues(); // Get form values
+  const avatarUrl = inputvalues.avatar;
+
+  if (!avatarUrl) {
+    return;
+  }
+
+  popupWithEditAvatarForm.setLoadingText(true);
+  api
+    .updateProfileAvatar(avatarUrl)
+    .then((res) => {
+      userInfo.setUserAvatar(res.avatar);
+      popupWithEditAvatarForm.close();
+    })
+    .catch((err) => console.error("Profile update failed:", err))
+    .finally(() => {
+      popupWithEditAvatarForm.setLoadingText(false);
     });
-    popupWithEditProfileForm.close();
-    editFormValidator.disableSubmitButton();
+
+  updateFormValidator.disableSubmitButton();
```

Comment:
`  updateFormValidator.disableSubmitButton();
`

This should be in line 129 in `.then`

You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.